lib/widgets/nft_collection_bloc.dart

```diff
@@ -1,106 +1,193 @@
+// ignore_for_file: public_member_api_docs, sort_constructors_first
+import 'dart:async';
+
 import 'package:bloc/bloc.dart';
 import 'package:collection/collection.dart';
+import 'package:flutter/foundation.dart';
+
 import 'package:nft_collection/database/nft_collection_database.dart';
+import 'package:nft_collection/models/address_index.dart';
 import 'package:nft_collection/models/asset_token.dart';
 import 'package:nft_collection/nft_collection.dart';
+import 'package:nft_collection/services/configuration_service.dart';
 import 'package:nft_collection/services/tokens_service.dart';
 import 'package:nft_collection/utils/constants.dart';
+import 'package:nft_collection/utils/list_extentions.dart';
 
 class NftCollectionBlocState {
-  static const _tokensEquality = ListEquality<AssetToken>();
-
   final NftLoadingState state;
   final List<AssetToken> tokens;
```

Comment:
as I mention yesterday, can we stripe all the unused fields to reduce memory consumption when retaining a big list of tokens in memory?